packages/flutter/lib/src/rendering/object.dart

```diff
@@ -15,6 +15,7 @@ import 'binding.dart';
 import 'debug.dart';
 import 'layer.dart';
 import 'node.dart';
+import 'proxy_box.dart' show RenderBlockSemantics;
```

Comment:
This import creates a circular dependency.  Although Dart can cope with circular dependencies, we avoid them because they tend to indicate a design flaw (e.g., the test against a specific subclass below).